cronjobs/ftp_connection.py

```diff
@@ -0,0 +1,88 @@
+# Python 3 module
+# -*- coding: utf-8 -*-
+from ftplib import FTP
+import os
+
+
+class FTPConnection:
+    _host = ""
+    _username = ""
+    _password = ""
+    _ftp = None
+
+    def __init__(self, host, username, password):
+        self._host = host
+        self._username = username
+        self._password = password
+        self._connect()
+        self._login()
+
+    def getFTP():
+        return _ftp
+
+    def _connect(self):
+        try:
+            self._ftp = FTP(host=self._host, timeout=120)
+        except Exception as e:
+            Error("failed to connect to FTP server! (" + str(e) + ")")
+
+    def _login(self):
+        try:
+            self._ftp.login(user=self._username, passwd=self._password)
+        except Exception as e:
+            Error("failed to login to FTP server! (" + str(e) + ")")
+
+    def changeDirectory(self, remote_dir_path):
+        error_message = "failed to change directory: " + remote_dir_path
+        try:
+            res = self._ftp.cwd(remote_dir_path)
+            if not res.startswith('250 '):
+                Error(error_message)
+        except Exception as e:
+            Error(error_message + " (" + str(e) + ")")
+
+    def listDirectory(self):
+        try:
+            return self._ftp.nlst()
+        except Exception as e:
+            Error("failed to list directory (" + str(e) + ")")
+
+    def downloadFile(self, remote_file_name, local_file_path=None):
+        if local_file_path is None:
+            local_file_path = remote_file_name
+        try:
+            output = open(local_file_path, "wb")
```

Comment:
FYI (no change necessary): On UNIX-like systems there is no difference between text and binary mode.  So, unless you want something to be portable to Windows the "b" is not necessary.  It's harmless and could be viewed as a hint to the reader that we intent to read or write binary files.

Reply:
Nice to know, ty!